detection_rules/schemas/__init__.py

```diff
@@ -238,4 +239,5 @@ def get_stack_schemas(stack_version: str) -> Dict[str, dict]:
     if stack_version > current_package:
         versions[stack_version] = {'beats': 'main', 'ecs': 'master'}
 
-    return versions
+    versions_reversed = OrderedDict(sorted(versions.items(), reverse=True))
```

Comment:
this just reverses the order of the validation to newest to oldest, so that the first failure is the first incompatible package (and an indicator to potentially add a min stack version)